habitat-lab/habitat/tasks/rearrange/multi_task/task_creator_utils.py

```diff
@@ -43,21 +45,25 @@ def create_task_object(
     task_cls = registry.get_task(task_cls_name)
 
     config = copy.deepcopy(cur_config)
-    with habitat.config.read_write(config):
-        if task_config_path is not None:
-            pass_args: List[Any] = []
-            for k, v in task_config_args.items():
-                pass_args.extend((k, v))
-            task_config = habitat.get_config(
-                osp.join(TASK_CONFIGS_DIR, task_config_path + ".yaml"),
-                pass_args,
-            )
+
+    if task_config_path is not None:
+        pass_args: List[Any] = []
+        for k, v in task_config_args.items():
+            pass_args.append(f"{k}={v}")
+        task_config = habitat.get_config(
+            osp.join(TASK_CONFIGS_DIR, task_config_path + ".yaml"),
+            pass_args,
+        )
+
+        with habitat.config.read_write(config):
+            config = OmegaConf.merge(config, task_config.habitat.task)
+            # config.merge_from_other_cfg(task_config.habitat.task)
+            # Putting back the values from TASK_IGNORE_KEYS :
             for k in TASK_IGNORE_KEYS:
```

Comment:
@ASzot I changed the way the `TASK_IGNORE_KEYS` are handled. Instead of removing them from the overriding config, I re-add them from the original base configuration. Let me know if this is not the behavior that was intended.